src/Microsoft.Data.SqlClient/tests/ManualTests/TracingTests/EventCounterTest.cs

```diff
@@ -16,91 +15,247 @@ namespace Microsoft.Data.SqlClient.ManualTesting.Tests
     /// </summary>
     public class EventCounterTest
     {
+        public EventCounterTest()
+        {
+            ClearConnectionPools();
+        }
+
+        [ConditionalFact(typeof(DataTestUtility), nameof(DataTestUtility.AreConnStringsSetup))]
+        public void EventCounter_HardConnectionsCounters_Functional()
+        {
+            //create a non-pooled connection
+            var stringBuilder = new SqlConnectionStringBuilder(DataTestUtility.TCPConnectionString) {Pooling = false};
+
+            using var conn = new SqlConnection(stringBuilder.ToString());
+
+            //initially we have no open physical connections
+            Assert.Equal(0, SqlClientEventSourceProps.ActiveHardConnections);
+            Assert.Equal(0, SqlClientEventSourceProps.NonPooledConnections);
+
+            conn.Open();
+
+            //when the connection gets opened, the real physical connection appears
+            Assert.Equal(1, SqlClientEventSourceProps.ActiveHardConnections);
+            Assert.Equal(1, SqlClientEventSourceProps.NonPooledConnections);
+
+            conn.Close();
+
+            //when the connection gets closed, the real physical connection is also closed
+            Assert.Equal(0, SqlClientEventSourceProps.ActiveHardConnections);
+            Assert.Equal(0, SqlClientEventSourceProps.NonPooledConnections);
+        }
+
         [ConditionalFact(typeof(DataTestUtility), nameof(DataTestUtility.AreConnStringsSetup))]
-        public void EventCounterTestAll()
+        public void EventCounter_SoftConnectionsCounters_Functional()
         {
-            var stringBuilder = new SqlConnectionStringBuilder(DataTestUtility.TCPConnectionString)
+            //create a pooled connection
+            var stringBuilder = new SqlConnectionStringBuilder(DataTestUtility.TCPConnectionString) {Pooling = true};
+
+            using (var conn = new SqlConnection(stringBuilder.ToString()))
             {
-                Pooling = true,
-                MaxPoolSize = 20
-            };
+                //initially we have no open physical connections
+                Assert.Equal(0, SqlClientEventSourceProps.ActiveHardConnections);
+                Assert.Equal(0, SqlClientEventSourceProps.ActiveSoftConnections);
+                Assert.Equal(0, SqlClientEventSourceProps.PooledConnections);
+                Assert.Equal(0, SqlClientEventSourceProps.NonPooledConnections);
+                Assert.Equal(0, SqlClientEventSourceProps.ActiveConnectionPools);
+                Assert.Equal(0, SqlClientEventSourceProps.ActiveConnections);
+                Assert.Equal(0, SqlClientEventSourceProps.FreeConnections);
+
+                conn.Open();
+
```

Comment:
Similar to the test case above this one. Can you also add the check for 
`_hardConnectsCounter - _hardDisconnectsCounter == _activeHardConnectionsCounter`
and 
`_softConnectsCounter - _softDisconnectsCounter == _activeSoftConnectionsCounter`
after we open and close and reopen connection?

Reply:
Done